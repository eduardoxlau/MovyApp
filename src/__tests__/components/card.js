/* eslint-disable no-undef */
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';

// eslint-disable-next-line import/no-unresolved
import moviesMock from '__mocks__/movies';
import Card from 'components/card';

describe('<Card />', () => {
  it('should render snapshot card wide', () => {
    const tree = renderer
      .create(<Card item={moviesMock[0]} type="wide" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render snapshot card large', () => {
    const tree = renderer
      .create(<Card item={moviesMock[0]} type="large" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should handler when clicked card', () => {
    const onSelected = jest.fn();
    const { getByTestId } = render(
      <Card item={moviesMock[0]} onSelected={onSelected} />
    );
    const card = getByTestId('card');
    fireEvent.click(card);
    expect(onSelected).toHaveBeenCalled();
  });
  it('should show up triangle icon when card is selected', () => {
    const { getByTestId } = render(
      <Card item={moviesMock[0]} idSelected={moviesMock[0].id} border />
    );
    const image = getByTestId('border-active');

    expect(image).toBeInTheDocument();
  });
});
