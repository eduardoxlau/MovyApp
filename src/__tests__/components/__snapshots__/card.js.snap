// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render snapshot card large 1`] = `
<div
  data-testid="card"
  onClick={[Function]}
>
  <div
    className="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt=""
      src="http://localhost:4000/view1.png"
    />
    <div
      className="absolute bottom-20 w-full flex justify-center"
    >
      <div
        className="bg-blue-600 px-4 font-bold text-sm"
      >
        MOVIE ORIGINAL
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card /> should render snapshot card wide 1`] = `
<div
  data-testid="card"
  onClick={[Function]}
>
  <div
    className="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
  >
    <img
      alt=""
      className="img-card"
      src="http://localhost:4000/card1.png"
    />
  </div>
</div>
`;
